File: lib.database/src/main/java/com/prance/lib/database/DaoHelper.java
package com.prance.lib.database;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;

import org.greenrobot.greendao.database.Database;

public class DaoHelper extends DaoMaster.OpenHelper {

    public DaoHelper(Context context, String name, SQLiteDatabase.CursorFactory factory) {
        super(context, name, factory);
    }

    /**
     * 更新数据库
     *
     * @param db
     * @param oldVersion
     * @param newVersion
     */
    @Override
    public void onUpgrade(Database db, int oldVersion, int newVersion) {

    }

}

File: lib.database/src/main/java/com/prance/lib/database/DaoHelper.kt
package com.prance.lib.database

import android.content.Context
import android.database.sqlite.SQLiteDatabase

import org.greenrobot.greendao.database.Database

class DaoHelper(context: Context, name: String, factory: SQLiteDatabase.CursorFactory?) : DaoMaster.OpenHelper(context, name, factory) {

    /**
     * 更新数据库
     *
     * @param db
     * @param oldVersion
     * @param newVersion
     */
    override fun onUpgrade(db: Database?, oldVersion: Int, newVersion: Int) {
        if (oldVersion <= 1) {
            MessageEntityDao.createTable(db, true)
        }
    }

}
